packages/lib/modules/lbp/steps/sale-structure/ProjectedPriceChart.tsx

```diff
@@ -0,0 +1,188 @@
+import { addHours, differenceInDays } from 'date-fns'
+import ReactECharts, { EChartsOption } from 'echarts-for-react'
+import * as echarts from 'echarts/core'
+import { bn } from '@repo/lib/shared/utils/numbers'
+
+export function ProjectedPriceChart({
+  startWeight,
+  endWeight,
+  startDate,
+  endDate,
+  launchTokenSeed,
+  collateralTokenSeed,
+  collateralTokenPrice,
+  onPriceChange,
+}: {
+  startWeight: number
+  endWeight: number
+  startDate: Date
+  endDate: Date
+  launchTokenSeed: number
+  collateralTokenSeed: number
+  collateralTokenPrice: number
+  onPriceChange: (prices: number[][]) => void
+}) {
+  const priceData = interpolateData(
+    startWeight,
+    endWeight,
+    startDate,
+    endDate,
+    launchTokenSeed,
+    collateralTokenSeed,
+    collateralTokenPrice
+  )
+
+  setTimeout(() => onPriceChange(priceData))
+
+  const priceRange = range(priceData.map(point => point[1]))
+
+  const chartInfo: EChartsOption = {
+    xAxis: {
+      show: true,
+      type: 'value',
+      axisLine: { show: false },
+      splitLine: { show: false },
+      axisTick: { show: false },
+      min: startDate.getTime(),
+      max: endDate.getTime(),
+      interval: 24 * 60 * 60 * 1000,
+      axisLabel: {
+        formatter: (value: number) => {
+          const daysDiff = differenceInDays(new Date(value), startDate)
+          return `Day ${daysDiff}`
+        },
+      },
+    },
+    yAxis: {
+      show: true,
+      type: 'value',
+      axisLine: { show: false },
+      splitLine: { show: false },
+      axisTick: { show: false },
+      axisLabel: {
+        formatter: (value: number) => {
+          return `$${value}`
+        },
+      },
+    },
+    series: [
+      {
+        id: 'top-markline',
+        type: 'line',
+        data: [
+          [startDate, priceRange.max],
+          [endDate, priceRange.max],
+        ],
+        lineStyle: {
+          color: 'grey',
+          type: 'dashed',
+          width: 1,
+          cap: 'round' as const,
+          join: 'round' as const,
+        },
+        showSymbol: false,
+      },
+      {
+        id: 'middle-markline',
+        type: 'line',
+        data: [
+          [startDate, (priceRange.max - 0) / 2],
+          [endDate, (priceRange.max - 0) / 2],
+        ],
+        lineStyle: {
+          color: 'grey',
+          type: 'dashed',
+          width: 1,
+          cap: 'round' as const,
+          join: 'round' as const,
+        },
+        showSymbol: false,
+      },
+      {
+        id: 'bottom-markline',
+        type: 'line',
+        data: [
+          [startDate, 0],
+          [endDate, 0],
+        ],
+        lineStyle: {
+          color: 'grey',
+          type: 'dashed',
+          width: 1,
+          cap: 'round' as const,
+          join: 'round' as const,
+        },
+        showSymbol: false,
+      },
```

Comment:
you could also add the mark lines by adding the following object to the `price` line serie:

```
        markLine: {
          data: [
            {
              yAxis: 0,
            },
            {
              yAxis: (priceRange.max - 0) / 2,
            },
            {
              yAxis: priceRange.max,
            },
          ],
          lineStyle: {
            color: 'grey',
            type: 'dashed',
            width: 1,
            cap: 'round',
            join: 'round',
          },
          symbol: 'none',
          silent: true,
          label: {
            show: false,
          },
```

(are the `as const`'s needed?)